resources/rewrite.rb

```diff
@@ -16,102 +16,158 @@
+
+property :value, String,
+          description: 'Value to apply rewrite action to (Field name)'
+
+property :values, [String, Array],
+          description: 'Values to apply rewrite action to (Field name or Glob pattern)'
 
+property :flags, [String, Array],
+          description: 'Flag(s) to apply'
+
+property :tags, String,
+          description: 'Tags to apply'
+
+property :condition, String,
+          description: 'Condition which must be satisfied for the rewrite to be applied'
+
+property :additional_options, Hash,
+          default: {},
+          description: 'Additional options for rewrite function'
+
+property :configuration, [Hash, Array],
+          coerce: proc { |p| p.is_a?(Array) ? p : [p] },
+          description: 'Hash or Array of Hash containing raw rewrite(s) configuration'
+
+action_class do
+  def rewrite_config
+    if new_resource.configuration
+      new_resource.configuration
+    else
+      case new_resource.function
+      when 'subst'
+        [
+          {
+            'function' => new_resource.function,
+            'match' => new_resource.match,
+            'replacement' => new_resource.replacement,
+            'value' => new_resource.value,
+            'flags' => new_resource.flags,
+            'condition' => new_resource.condition,
+            'additional_options' => new_resource.additional_options,
+          },
+        ]
+      when 'set'
+        [
+          {
+            'function' => new_resource.function,
+            'replacement' => new_resource.replacement,
+            'value' => new_resource.value,
+            'condition' => new_resource.condition,
+            'additional_options' => new_resource.additional_options,
+          },
+        ]
+      when 'unset groupunset'
+        [
+          {
+            'function' => new_resource.function,
+            'field' => new_resource.field,
+            'value' => new_resource.value,
+            'values' => new_resource.values,
+            'condition' => new_resource.condition,
+            'additional_options' => new_resource.additional_options,
+          },
+        ]
+      when 'groupset'
+        [
+          {
+            'function' => new_resource.function,
+            'field' => new_resource.field,
+            'values' => new_resource.values,
+            'condition' => new_resource.condition,
+            'additional_options' => new_resource.additional_options,
+          },
+        ]
+      when 'set-tag clear-tag'
+        [
+          {
+            'function' => new_resource.function,
+            'tags' => new_resource.tags,
+          },
+        ]
+      when 'credit-card-mask'
+        [
+          {
+            'function' => new_resource.function,
+            'value' => new_resource.value,
+            'condition' => new_resource.condition,
+            'additional_options' => new_resource.additional_options,
+          },
+        ]
+      end
+    end
+  end
+end
```

Comment:
What do you think about extracting this into it's own helper?

Reply:
Ah! This has slipped through the net I was planning on doing exactly that.